purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -302,23 +362,21 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         listener: ReceivePurchaserInfoListener
     ) {
         debugLog("Restoring purchases")
-        billingWrapper.queryPurchaseHistoryAsync(
-            BillingClient.SkuType.SUBS,
-            { subsPurchasesList ->
-                billingWrapper.queryPurchaseHistoryAsync(
-                    BillingClient.SkuType.INAPP,
-                    { inAppPurchasesList ->
-                        val allPurchases = ArrayList(subsPurchasesList)
-                        allPurchases.addAll(inAppPurchasesList)
-                        if (allPurchases.isEmpty()) {
-                            getPurchaserInfo(listener)
-                        } else {
-                            postRestoredPurchases(allPurchases, listener)
-                        }
-                    },
-                    { error -> dispatch { listener.onError(error) } })
-            },
-            { error -> dispatch { listener.onError(error) } })
+        if (!allowSharingPlayStoreAccount) {
+            debugLog("allowSharingPlayStoreAccount is set to false and restorePurchases has been called. Are you sure you want to do this?")
```

Comment:
"allowSharingPlayStoreAccount is set to false and restorePurchases has been called. This will 'alias' any app user id's sharing the same receipt. Are you sure you want to do this?"